Data Fetching from Airtable

MilaKova opened the pull request.

Commit:
Data Fetching fro Airtable

GinaCastromonte reviewed (changes requested):
On your AddTodoForm.js file,  it appears that you're trying to call the onAddTodo function and then define an object {} right after without proper separation or structure.To fix the syntax error, modify the handleAddTodo function within the AddTodoForm component to correctly pass the todo information as an object to the onAddTodo function.  `const handleAddTodo = (e) => {
  e.preventDefault();
  const newTodo = {
    id: Date.now(),
    title: todoTitle,
  };
  onAddTodo(newTodo); 
  setTodoTitle("");
};
`

Also I took out your postTodo function and bakctracked.  This code seems to be working correctly.  I would go back and try a POST route again once you can see your local storage is working and connected to the API.  `import React, {useState, useEffect} from 'react';
import TodoList from'./TodoList.js';
import AddTodoForm from './AddTodoForm';


function App() {

  const [todoList, setTodoList]= useState([]);
  const [isLoading, setIsLoading] = useState(true);
  const [isError, setIsError] = useState(false);




  const fetchData = async() => {
    const options = 
    {
      method: "GET",
      headers: {
        'Authorization': `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`
      },
    }
    const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`

    try {
      const response = await fetch (url, options);
    
      if (!response.ok) {
        const message = `Error: ${response.status}`
        throw new Error(message);
      }

      const data = await response.json();
      console.log("data:", data);
      const todos= data.records.map((todo)=>{
        const newTodo = {
          id: todo.id,
          title: todo.fields.title,
          completedAt: todo.createdTime,             
        }
        return newTodo;          
      });
      console.log("todos:", todos);

      setTodoList(todos);
      setIsLoading(false);
      setIsError(false);

    } catch (error){
      console.log(error.message);
      };         
  }; 


  useEffect(() => {
    fetchData()
  },[]);

  useEffect(() => {
    if (isLoading === false) 
    localStorage.setItem('savedTodoList', JSON.stringify(todoList))
  },[todoList, isLoading])

  const removeTodo = (id)=> {
    const newTodo = todoList.filter(
      (todoList)=>id !==todoList.id
    )
   setTodoList(newTodo);
  };

  const addTodo = (newTodo) => {
    setTodoList([...todoList, newTodo])
  };

  // const postTodo = async (todo) => {
  //   try {
  //     const airtableData = {
  //       fields: {
  //       title: todo,
  //       },
  //     };
  //     console.log("airtableData:", airtableData)

  //     const response = await fetch(
  //       `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`,

  //       {
  //         method: "POST",
  //         headers: {
  //           "Content-Type": "application/json",
  //           Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
  //         },
  //         body: JSON.stringify(airtableData),
  //       }
  //     );
      
  //     if (!response.ok) {
  //       const message = `Error: ${response.status}`;
  //       throw new Error(message);
  //     }

  //     const responseData = await response.json();
     
  //       const newTodo = {
  //        id:responseData.id,
  //        title:responseData.fields.title,
  //       }
     
  //     setTodoList([...todoList, newTodo]);

  //   } catch (error) {
  //     console.log(error.message);
  //     return null;
  //     }  
  // };

  return (
    <>  
      <h1>Todo list</h1>      
      <AddTodoForm onAddTodo={addTodo}/>

      {isError && <p>Something went wrong</p>}

      {isLoading ? (
        <p>Loading...</p>
      ) : (
      <TodoList todoList={todoList} onRemoveTodo={removeTodo}/>
      )}
    </>
  );
};

export default App;
`

Commit:
Added POST and DELETE methods

MilaKova commented:
Thank you, Gina, for review and comments. I fixed AddTodoForm.js file, as you suggested, but still confused if we need to add Date.now() function or not if we use POST feature.
Also I corrected addTodo function and figured out how to make removeTodo, added it as well. 
For me now everything works as expected.

MilaKova committed:
Merge branch 'main' into lesson_1_8
